File: src/main/java/br/com/tech/challenge/service/TopicService.java
package br.com.tech.challenge.service;


import br.com.tech.challenge.definitions.Result;
import br.com.tech.challenge.exceptions.TopicNotFoundException;
import br.com.tech.challenge.model.Topic;
import br.com.tech.challenge.repository.TopicRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@AllArgsConstructor
public class TopicService {

    private final TopicRepository topicRepository;
    public Topic save(Topic topic) {
        return topicRepository.save(topic);
    }

    public Topic findById(String id) {
        return topicRepository.findById(UUID.fromString(id)).orElseThrow(() ->
        new TopicNotFoundException(STR."Agenda with {id} = \{id} not found"));
    }

}

File: src/main/java/br/com/tech/challenge/service/TopicService.kt
package br.com.ntconsult.challenge.service

import br.com.ntconsult.challenge.exception.TopicNotFoundException
import br.com.ntconsult.challenge.model.Topic
import br.com.ntconsult.challenge.repository.TopicRepository
import org.springframework.stereotype.Service
import java.util.*

@Service
class TopicService(private val topicRepository: TopicRepository) {

    fun save(topic: Topic): Topic {
        return topicRepository.save(topic)
    }

    fun findById(id: String): Topic {
        return topicRepository.findById(UUID.fromString(id)).orElseThrow {
            TopicNotFoundException("Agenda with {id} = $id not found")
        }
    }

}
